test/suites/scheduling/suite_test.go

```diff
@@ -85,6 +99,16 @@ var _ = Describe("Scheduling", Ordered, ContinueOnFailure, func() {
 			// TODO: review the use of "kubernetes.azure.com/cluster"
 			v1beta1.AKSLabelCluster,
 		)
+
+		// If no spec with Label("GPU") ran (e.g., `-label-filter='!GPU'`),
+		// ignore GPU labels in the coverage assertion.
+		if !gpuSpecsRan.Load() {
```

Comment:
Maybe just something like this?
```golang
if !Label("GPU").MatchesLabelFilter(GinkgoLabelFilter()) {
```

Reply:
didnt know that exists, seems much better